### PR TITLE
Persist sessions

### PR DESCRIPTION
**Pull request by FlorianBouron**

### CODE REVIEW

File: packages/extension/src/components/pages/CreatePassword/CreatePassword.tsx [hunk 1]
```diff
@@ -36,16 +42,25 @@ export const CreatePassword: FC<CreatePasswordProps> = ({
       setPasswordError('Passwords must match');
     } else {
       try {
-        saveWallet(wallet, confirmPasswordValue);
-        setActiveStep((prevActiveStep) => prevActiveStep + 1);
```
Comment (ThibautBremand) [inline packages/extension/src/components/pages/CreatePassword/CreatePassword.tsx:40]: this is removed?

**Comment by ThibautBremand** [comment]
The session is not destroyed when the user clicks on "Lock"

**Comment by ThibautBremand** [comment]
Locally my session expires quite fast (like 1 min)

**Comment by ThibautBremand** [comment]
@all-contributors add @qlemaire for security

File: packages/constants/src/message/message.types.ts [hunk 2]
```diff
@@ -243,6 +251,13 @@ export type MessagingResponse = {
   error?: MessagingError;
 };
 
+// Internal
+export type InternalMessagingResponse = {
+  source?: 'INTERNAL_RESPONSE';
+  messagedId?: number;
+  error?: MessagingError;
```
Comment (FlorianBouron) [inline packages/constants/src/message/message.types.ts:258]: Maybe let's remove this

File: packages/extension/src/components/pages/CreatePassword/CreatePassword.tsx (hunk 1 above)
Comment (FlorianBouron) [reply packages/extension/src/components/pages/CreatePassword/CreatePassword.tsx:40]: Yes, that was for testing purposes, thanks for reverting it :)

File: packages/constants/src/message/message.types.ts (hunk 2 above)
Comment (FlorianBouron) [reply packages/constants/src/message/message.types.ts:262]: We will keep it as this a a generic type.

File: packages/extension/src/chromeServices/background/index.ts [hunk 3]
```diff
@@ -45,11 +49,27 @@ import {
...
 import { focusOrCreatePopupWindow } from './utils/focusOrCreatePopupWindow';
+import { Session } from './utils/session';
 
 const sendMessageToTab = <T>(tabId: number | undefined, message: any) => {
   chrome.tabs.sendMessage<T>(tabId ?? 0, message);
 };
 
+async function createOffscreen() {
+  await chrome.offscreen
+    .createDocument({
```
Comment (ThibautBremand) [inline packages/extension/src/chromeServices/background/index.ts:60]: We should also close the document when the session expires, and only open it when a session is created

File: packages/extension/src/chromeServices/background/utils/session.ts [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+export class Session {
+  private static instance: Session | null = null;
+  private password: string | null = null;
+  private timeoutId: NodeJS.Timeout | null = null;
+
+  private constructor() {}
+
+  public static getInstance(): Session {
+    if (!Session.instance) {
+      Session.instance = new Session();
```
Comment (ThibautBremand) [inline packages/extension/src/chromeServices/background/utils/session.ts:13]: To be checked

**Pull request merged**